File: 2540-minimum-common-value/2540-minimum-common-value.cpp
class Solution {
public:
    int getCommon(vector<int>& nums1, vector<int>& nums2) {
        int i = 0, j = 0;
        int len1 = nums1.size();
        int len2 = nums2.size();

        while (i < len1 && j < len2) {
            if (nums1[i] == nums2[j]) {
                return nums1[i];
            } else if (nums1[i] < nums2[j]) {
                i++;
            } else {
                j++;
            }
        }

        return -1;
    }
};

File: 2540-minimum-common-value/2540-minimum-common-value.java
class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        int ans = -1;
        for (int i1 = 0, i2 = 0; i1 < nums1.length && i2 < nums2.length; ) {
            int n1 = nums1[i1];
            int n2 = nums2[i2];

            if (n1 == n2) {
                ans = n1;
                break;
            }

            if (n1 < n2) {
                i1 += 1;
            } else {
                i2 += 1;
            }
        }

        return ans;
    }
}